contracts/FindUserStatus.cdc

```diff
@@ -0,0 +1,296 @@
+							)
+						}
+						listingsV2 = StorefrontListing(storefrontID: d.storefrontID, nftType: d.nftType.identifier, nftID: d.nftID, salePaymentVaultType: d.salePaymentVaultType.identifier, salePrice: d.salePrice, saleCuts: saleCuts, customID: d.customID, commissionAmount: d.commissionAmount, expiry: d.expiry)
+						return listingsV2
+					}
+				}
+			}
+		}
+		return nil
+	}
+
+	pub fun getFlowtyListing(user: Address, id: UInt64) : FlowtyListing? {
+		var flowty : FlowtyListing? = nil
+		let account = getAccount(user)
+		let flowtyCap = account.getCapability<&Flowty.FlowtyStorefront{Flowty.FlowtyStorefrontPublic}>(Flowty.FlowtyStorefrontPublicPath)
+
+		if flowtyCap.check() {
+			let storefrontRef=flowtyCap.borrow()!
+			for listingId in storefrontRef.getListingIDs() {
+				let listing = storefrontRef.borrowListing(listingResourceID: listingId)!
+				let d = listing.getDetails()
+				if d.nftID==id {
+					if !d.funded {
+						let saleCuts : [StoreFrontCut] = [] 
+						for cut in d.getPaymentCuts() {
+							saleCuts.append(
+								StoreFrontCut(
+									amount: cut.amount, 
+									address: cut.receiver.address
+								)
+							)
+						}
+						flowty = FlowtyListing(flowtyStorefrontID: d.flowtyStorefrontID, funded: d.funded, nftType: d.nftType.identifier, nftID: d.nftID, amount: d.amount, interestRate: d.interestRate, term: d.term, paymentVaultType: d.paymentVaultType.identifier ,paymentCuts: saleCuts, listedTime: d.listedTime, royaltyRate: d.royaltyRate, expiresAfter: d.expiresAfter, repaymentAmount: d.getTotalPayment())
+						return flowty
+					}
+				}
+			}
+		}
+		return nil
+	}
+
+	pub fun getFlowtyRentals(user: Address, id: UInt64) : FlowtyRental? {
+
+		var flowtyRental : FlowtyRental? = nil
+		let account = getAccount(user)
+		let flowtyRentalCap = account.getCapability<&FlowtyRentals.FlowtyRentalsStorefront{FlowtyRentals.FlowtyRentalsStorefrontPublic}>(FlowtyRentals.FlowtyRentalsStorefrontPublicPath)
+
+		if flowtyRentalCap.check() {
+			let storefrontRef=flowtyRentalCap.borrow()!
+			for listingId in storefrontRef.getListingIDs() {
+				let listing = storefrontRef.borrowListing(listingResourceID: listingId)!
+				let d = listing.getDetails()
+				if d.nftID==id {
+					if !d.rented {
+						let saleCuts : [StoreFrontCut] = [] 
+						let cut = d.getPaymentCut() 
+							saleCuts.append(
+								StoreFrontCut(
+									amount: cut.amount, 
+									address: cut.receiver.address
+								)
+							)
+						
+						flowtyRental = FlowtyRental(flowtyStorefrontID: d.flowtyStorefrontID, rented: d.rented, nftType: d.nftType.identifier, nftID: d.nftID, amount: d.amount, deposit: d.deposit, term: d.term, paymentVaultType: d.paymentVaultType.identifier, reenableOnReturn: d.reenableOnReturn, paymentCuts: saleCuts, listedTime: d.listedTime, royaltyRate: d.royaltyRate, expiresAfter: d.expiresAfter, repaymentAmount: d.getTotalPayment(), renter: d.renter)
+						return flowtyRental
+					}
+				}
+			}
+		}
+		return nil
+	}
```

Comment:
I think we should add flovatar market here, it is a pretty popular market for flovatar and if a flovatar is listed for sale I think it should be in our nft details page.